sign_oca/models/sign_oca_request.py

```diff
@@ -76,6 +76,19 @@ def _compute_next_item_id(self):
                 record.signatory_data and max(record.signatory_data.keys()) or 0
             ) + 1
 
+    def preview(self):
+        self.ensure_one()
```

Comment:
Might be able to add a check if the user is a signer and the document is not signed, and then allow him to sign?

Reply:
In this fix #18, a button to allow sign from the request form has been added. 

![image](https://github.com/OCA/sign/assets/137763287/a4ffea1e-fd78-495e-8256-e3a9e01b26b7)

This button is only available if the user has to sign the request. Is it solving your request?